src/components/Tree/helpers/getMovementAnnouncements.ts

```diff
@@ -0,0 +1,95 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { Dispatch, SetStateAction } from 'react';
+import { arrayMove } from '@dnd-kit/sortable';
+
+import { TreeState } from '../types';
+
+type AnnouncementItem = {
+    level: number;
+    id: string;
+    parentId: string;
+};
+
+type AnnouncementArgs = {
+    eventName: string;
+    activeId: string;
+    overId?: string;
+    treeState: TreeState;
+    currentPosition: Nullable<{ overId: string; parentId: Nullable<string> }>;
+    setCurrentPosition: Dispatch<
+        SetStateAction<
+            Nullable<{
+                overId: string;
+                parentId: Nullable<string>;
+            }>
+        >
+    >;
+};
+
+export const getMovementAnnouncement = ({
+    eventName,
+    activeId,
+    overId,
+    treeState,
+    currentPosition,
+    setCurrentPosition,
+}: AnnouncementArgs) => {
+    const projected = treeState.projection;
+
+    if (overId && projected) {
+        if (eventName !== 'onDragEnd') {
+            if (
+                currentPosition &&
+                projected.parentId === currentPosition.parentId &&
+                overId === currentPosition.overId
+            ) {
+                return;
+            } else {
+                setCurrentPosition({
+                    parentId: projected.parentId,
+                    overId,
+                });
+            }
+        }
+
+        const announcementNodes: AnnouncementItem[] = treeState.nodes.map(({ props }) => ({
+            id: props.id,
+            level: props.level,
+            parentId: props.parentId,
+        }));
+
+        const overIndex = announcementNodes.findIndex(({ id }) => id === overId);
+        const activeIndex = announcementNodes.findIndex(({ id }) => id === activeId);
+        const sortedItems = arrayMove(announcementNodes, activeIndex, overIndex);
+
+        const previousItem = sortedItems[overIndex - 1];
+
+        let announcement;
+        const movedVerb = eventName === 'onDragEnd' ? 'dropped' : 'moved';
+        const nestedVerb = eventName === 'onDragEnd' ? 'dropped' : 'nested';
+
+        if (!previousItem) {
+            const nextItem = sortedItems[overIndex + 1];
+            announcement = `${activeId} was ${movedVerb} before ${nextItem.id}.`;
```

Comment:
Same in this file the ID should be the text content first and fallback to id if there is no text